sig-and-wg/sig-core/proposals/objects-store-proposal.md

```diff
@@ -0,0 +1,155 @@
+
+You might want to use different bucket per solution. This is why you need to be able to specify multiple buckets in the ObjectStore. For example one bucket for documentation and one bucket per Web application.
+
+Another use cases for having multiple Buckets configuration:
+- Future extensibility by alowing bucket policy specification per solution
+- Minio doesn't support setting a bucket to behave as a static website host. Future controller of the Bucket custom resource will have to handle this additional functionality
+- Minio doesn't support setting a CDN for your objects. Future controller of the Bucket custom resource will have to handle this additional functionality. For example controller will be reesponsible for configuring a CloudFront for your bucket on S3
+
+Example resource for first version of the ObjectStore:
+```
+apiVersion: bucket.objectstore.kyma-project.io/v1alpha1
+kind: Bucket
+metadata:
+  name: my-bucket
+  namespace: stage
+#ObjectStore 2.0  
+#spec:
+#  policy: public #or other policies
+status:
+  ready: False
+  reason: BucketCreationFailure
+  message: "service unavailable"
+```
+
+You reference the Bucket in your Object CR with the following info in the spec:
+```
+  bucketRef:
+    name: my-bucket
+```
+
+It must be provided because the Object controller checks the Bucket custom resource status to make sure the bucket exists.
+
+## Object custom resource
+
+Object resource mandatory information is the:
+- reference info about the source file/object location that must be fetched by ObjectStore with 2 different modes:
+  - direct link fetch - the link points directly to file that need to be fetched
+  - index.yaml ref - the link to index file that contains reference to files that need to be separately fetch from a given relative location
+```
+#sample of index file with markdown and assets files
+apiVersion: v1
+files:
+  - name: 01-overview.md
+    metadata:
+      title: MyOverview
+      type: Overview
+  - name: 02-details.md
+    metadata:
+      title: MyDetails
+      type: Details
+  - name: 03-installation.md
+    metadata:
+      title: MyInstallation
+      type: Tutorial
+  - name: assets/diagram.svg
+```
+- reference to the bucket where the object should be stored
+
+
+```
+apiVersion: object.objectstore.kyma-project.io/v1alpha1
+kind: Object
+metadata:
+  name: my-package-objects
+spec:
+  source:
+    package: https://some.domain.com/my.zip
+  bucketRef:
+    name: my-bucket
+---
+apiVersion: objectstore.kyma-project.io/v1alpha1
+kind: Object
+metadata:
+  name: my-indexbased-objects
+spec:
+  source:
+    index: https://some.domain.com/index.yaml
+    rewrites:
+      - regex: 
+          find: \stitle="(.*)?"\s*(/>*)
+          replace: $2<title>$1</title>
+  filesFrom:
+    - configMapRef:
+        name: additional-object
+    - configMapRef:
+        name: one-more-object
+  bucketRef:
+    name: my-bucket
+status:
+  ready: True
+  objects:
+    - url: https://some.storage.domain/01-overview.md
```

Comment:
How will the user know which files from the index.yaml map to which objects? based on metadata title and type? does that mean title and type should be unique? 

An easier alternative is to specify the exposed base URI under which the files will be exposed in a path structure similar to the index.yaml. 
e.g.

```
  baseURI: https://some.storage.domain/
  objects:
    - url: https://some.storage.domain/01-overview.md
      metadata:
        title: MyOverview
        type: Overview
    - url: https://some.storage.domain/02-details.md
      metadata:
        title: MyDetails
        type: Details
```

Reply:
the uniqueness/mapping is achieved on `01-overview.md` and `02-details.md` parts of the URL. Filenames are not modified. Metadata can be anything, no uniqueness there.

I like your baseURI idea, but then should it look something like:
```
baseURI: https://some.storage.domain/
  objects:
    - name: 01-overview.md
       metadata:
         title: MyOverview
         type: Overview
    - name: 02-details.md
       metadata:
         title: MyDetails
         type: Details
```